uniswap-v2-core/contracts/UniswapV2Pair.sol

```diff
@@ -85,20 +118,31 @@ contract UniswapV2Pair is IUniswapV2Pair, UniswapV2ERC20 {
         emit Sync(reserve0, reserve1);
     }
 
-    // if fee is on, mint liquidity equivalent to 1/6th of the growth in sqrt(k)
+    function _calcFee(uint _sqrtNewK, uint _sqrtOldK, uint _platformFee, uint _circulatingShares) internal pure returns (uint _sharesToIssue) {
+        // Assert newK & oldK        < uint112
+        // Assert _platformFee       < FEE_ACCURACY
+        // Assert _circulatingShares < uint112
+
+        uint256 _scaledGrowth = (_sqrtNewK * ACCURACY) / _sqrtOldK;                      // ASSERT: < UINT256
+        uint256 _scaledMultiplier = ACCURACY - (SQUARED_ACCURACY / _scaledGrowth);          // ASSERT: < UINT128
+        uint256 _scaledTargetOwnership = _scaledMultiplier * _platformFee / FEE_ACCURACY; // ASSERT: < UINT144 during maths, ends < UINT128
+
+        _sharesToIssue = _scaledTargetOwnership.mul(_circulatingShares) / (ACCURACY.sub(_scaledTargetOwnership)); // ASSER: _scaledTargetOwnership < ACCURACY
+    }
+
     function _mintFee(uint112 _reserve0, uint112 _reserve1) private returns (bool feeOn) {
-        address feeTo = IUniswapV2Factory(factory).feeTo();
-        feeOn = feeTo != address(0);
-        uint _kLast = kLast; // gas savings
+        feeOn = platformFee > 0;
+        uint _kLast = kLast;
+
         if (feeOn) {
-            if (_kLast != 0) {
-                uint rootK = Math.sqrt(uint(_reserve0).mul(_reserve1));
-                uint rootKLast = Math.sqrt(_kLast);
-                if (rootK > rootKLast) {
-                    uint numerator = totalSupply.mul(rootK.sub(rootKLast));
-                    uint denominator = rootK.mul(5).add(rootKLast);
-                    uint liquidity = numerator / denominator;
-                    if (liquidity > 0) _mint(feeTo, liquidity);
+            address feeTo = IUniswapV2Factory(factory).feeTo();
+            uint _sqrtNewK = Math.sqrt(uint(_reserve0).mul(_reserve1));
+            uint _sqrtOldK = Math.sqrt(kLast); // gas savings
```

Comment:
Confirm no stack to deep issue from these new variables

Reply:
This should be verified through solc compilation I think? Or do we need to cover with additional testing as well?

Reply:
Ahh, yes - you appear to be right. This should absolutely be caught at compile time and online discourse seems to indicate that. If not then, im sure we hit all paths in tests?